### PR TITLE
Add mkTransposedPerSystemModule for simple per-system exposed attrs

### PR DESCRIPTION
**Pull request by shlevy**

### CODE REVIEW

**Comment by shlevy** [comment]
Note that docs won't properly render until flake-parts's doc generation is up to date with the mdDoc changes.

**Comment by shlevy** [comment]
@roberth Any thoughts on this?

File: lib.nix [hunk 1]
```diff
@@ -98,6 +98,34 @@ let
         type = flake-parts-lib.mkPerSystemType module;
       };
 
+    # Helper function for defining a per-system option that
+    # gets transposed by the usual flake system logic to a
+    # top-level flake attribute.
+    mkPerSystemModule = { name, option, file }: {
```
Comment (roberth) [inline lib.nix:104]:
```suggestion
    mkTransposedPerSystemModule = { name, option, file }: {
```

The transposition is quite relevant, as a perSystem module could be just one of those submodules.

It's also about a single attribute, but representing that seems to get too long.

File: dev/default.nix [hunk 2]
```diff
@@ -1,6 +1,6 @@
 let
   flake = builtins.getFlake (toString ./.);
-  fmc-lib = import ../lib.nix { inherit (flake.inputs.nixpkgs) lib; };
+  fmc-lib = (builtins.getFlake (toString ../.)).lib;
```
Comment (roberth) [inline dev/default.nix:3]: `..` started out as a lockless flake. (fwiw)

This looks ok.

**Review by roberth — COMMENTED** [review]
I didn't get around to this one, but it's not too complicated :)

File: lib.nix (hunk 1 above)
Comment (shlevy) [reply lib.nix]: Done.

**Review by roberth — APPROVED** [review]

**Pull request merged**